pyiceberg/table/update/snapshot.py

```diff
@@ -126,6 +128,11 @@ def __init__(
         self._deleted_data_files = set()
         self.snapshot_properties = snapshot_properties
         self._manifest_num_counter = itertools.count(0)
+        from pyiceberg.table import TableProperties
+
+        self._compression = self._transaction.table_metadata.properties.get(  # type: ignore
+            TableProperties.WRITE_AVRO_COMPRESSION, TableProperties.WRITE_AVRO_COMPRESSION_DEFAULT
+        )
```

Comment:
I think little things like table property parsing has gotten us unexpectedly in the past - would it be possible to add a few cases that demonstrate:
- that the newly written manifests now have the new default compression when a new snapshot is committed?
- and that newly written manifests respect a compression codec value when the property is set, when a new snapshot is committed?

I think this would help us add coverage for the new parameter both in property parsing and the simple logic here in `update.snapshots` module in each functions